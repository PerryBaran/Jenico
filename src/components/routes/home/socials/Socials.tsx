import React from 'react';
import style from './socials.module.css';

function Socials(props: {src: string, img:string, name: string}) {
    const {src, img, name} = props

    return (
        <a href={src} target="_Blank" rel="noreferrer" className={style.link}>
            <img alt={`${name} icon`} src={img}/>
            <p>{name}</p>
        </a>
    );
}

export default Socials;