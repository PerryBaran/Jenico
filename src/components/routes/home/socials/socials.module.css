.link {
    margin: 10px;
    height: 100px;
    text-shadow: 0px 0px 10px rgb(44, 0, 38);
    overflow: visible;
}

.link:hover > img{
    width: 55px;
}

.link img {
    display: flex;
    justify-content: center;
    align-items: center;

    width: 50px;
    
    transition-duration: 500ms;
}