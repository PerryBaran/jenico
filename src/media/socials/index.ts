import spotify from "./Spotify.png";
import bandcamp from "./BandCamp.png";
import soundcloud from "./Soundcloud.png";
import youtube from "./Youtube.png";
import instagram from "./Instagram.png";

class Social {
  readonly name: string;
  readonly src: string;
  readonly href: string;

  constructor(name: string, src: string, href: string) {
    this.name = name;
    this.src = src;
    this.href = href;
  }
}

const socials = [
  new Social(
    "Spotify",
    spotify,
    "https://open.spotify.com/artist/6XqSoJQMIWNKBuLzzmJwiz"
  ),
  new Social("Bandcamp", bandcamp, "https://jenico.bandcamp.com/"),
  new Social("Soundcloud", soundcloud, "https://soundcloud.com/jenico_flac"),
  new Social(
    "YouTube",
    youtube,
    "https://www.youtube.com/channel/UCOHuR2EY8wKXsDcocxDvqOA"
  ),
  new Social("Instagram", instagram, "https://www.instagram.com/jenico.flac/"),
];

export default socials;
