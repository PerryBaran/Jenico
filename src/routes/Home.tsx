import React from 'react';
import Background from '../components/Background';
import { background } from '../media/videos/index';
import { spotify, bandcamp, soundcloud, youtube, instagram} from '../media/icons/index';

function Home(props:any) {

    return (
        <div>
            <header className="home">
                <div>
                    <h1>Jenico</h1>
                    <div>
                        <a href="https://open.spotify.com/artist/6XqSoJQMIWNKBuLzzmJwiz" target="_Blank" rel="noreferrer"><img alt="spotify" src={spotify}/><p>Spotify</p></a>
                        <a href="https://jenico.bandcamp.com/" target="_Blank" rel="noreferrer"><img alt="bandcamp" src={bandcamp}/><p>Bandcamp</p></a>
                        <a href="https://soundcloud.com/jenico_flac" target="_Blank" rel="noreferrer"><img alt="soundcloud" src={soundcloud}/><p>SoundCloud</p></a>
                        <a href="https://www.youtube.com/channel/UCOHuR2EY8wKXsDcocxDvqOA" target="_Blank" rel="noreferrer"><img alt="youtube" src={youtube}/><p>YouTube</p></a>
                        <a href="https://www.instagram.com/jenico.flac/" target="_Blank" rel="noreferrer"><img alt="instagram" src={instagram}/><p>Instagram</p></a>
                    </div>
                </div>
                
            </header>
            <Background src={background}/>
        </div>
    );
}

export default Home;