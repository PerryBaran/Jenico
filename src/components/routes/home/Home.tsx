import React from 'react';
import BackgroundVideo from '../../background/video/BackgroundVideo';
import { background } from '../../../media/videos/index';
import { spotify, bandcamp, soundcloud, youtube, instagram} from '../../../media/icons/index';
import style from "./home.module.css";
import Socials from './socials/Socials';

function Home() {
    return (
        <div>
            <header className={style.container}>
                <div>
                    <h1>Jenico</h1>
                    <div>
                        <Socials src="https://open.spotify.com/artist/6XqSoJQMIWNKBuLzzmJwiz" img={spotify} name="Spotify"/>
                        <Socials src="https://jenico.bandcamp.com/" img={bandcamp} name="Bandcamp"/>
                        <Socials src="https://soundcloud.com/jenico_flac" img={soundcloud} name="Soundcloud"/>
                        <Socials src="https://www.youtube.com/channel/UCOHuR2EY8wKXsDcocxDvqOA" img={youtube} name="YouTube"/>
                        <Socials src="https://www.instagram.com/jenico.flac/" img={instagram} name="Instagram"/>
                    </div>
                </div>
            </header>
            <BackgroundVideo src={background}/>
        </div>
    );
}

export default Home;