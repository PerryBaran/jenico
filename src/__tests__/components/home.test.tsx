import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../../components/routes/home/Home";
import socials from "../../media/socials";

describe("Home", () => {
  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(<Home />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("content", () => {
    render(<Home />);

    expect(screen.getByText(/jenico/i)).toBeInstanceOf(HTMLHeadingElement);
    expect(screen.getByTestId("video")).toBeInTheDocument();

    const links = screen.getAllByRole("link");
    links.forEach((link, i) => {
      const social = socials[i];
      const image = screen.getByAltText(`${social.name} icon`);

      expect(link).toHaveAttribute("href", social.href);
      expect(image).toHaveAttribute("src", social.src);
      expect(screen.getByText(social.name)).toBeInTheDocument();
    });
  });
});