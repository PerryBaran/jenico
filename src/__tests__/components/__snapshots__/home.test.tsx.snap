// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<DocumentFragment>
  <header
    class="container"
  >
    <div>
      <h1>
        Jenico
      </h1>
      <div>
        <a
          class="social"
          href="https://open.spotify.com/artist/6XqSoJQMIWNKBuLzzmJwiz"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Spotify icon"
            src="Spotify.png"
          />
          <p>
            Spotify
          </p>
        </a>
        <a
          class="social"
          href="https://jenico.bandcamp.com/"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Bandcamp icon"
            src="BandCamp.png"
          />
          <p>
            Bandcamp
          </p>
        </a>
        <a
          class="social"
          href="https://soundcloud.com/jenico_flac"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Soundcloud icon"
            src="Soundcloud.png"
          />
          <p>
            Soundcloud
          </p>
        </a>
        <a
          class="social"
          href="https://www.youtube.com/channel/UCOHuR2EY8wKXsDcocxDvqOA"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="YouTube icon"
            src="Youtube.png"
          />
          <p>
            YouTube
          </p>
        </a>
        <a
          class="social"
          href="https://www.instagram.com/jenico.flac/"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Instagram icon"
            src="Instagram.png"
          />
          <p>
            Instagram
          </p>
        </a>
      </div>
    </div>
  </header>
  <div
    class="container"
  >
    <video
      autoplay=""
      class="hide"
      data-testid="video"
      loop=""
      src="DreamingOfDetunedLove.mp4"
    />
  </div>
</DocumentFragment>
`;
